File: src/main/java/cn/labzen/meta/exception/LabzenRuntimeException.java
package cn.labzen.meta.exception;

import org.slf4j.helpers.MessageFormatter;

public class LabzenRuntimeException extends RuntimeException {

  public LabzenRuntimeException(String message) {
    super(message);
  }

  public LabzenRuntimeException(String message, Object... args) {
    super(MessageFormatter.basicArrayFormat(message, args));
  }

  public LabzenRuntimeException(Throwable cause) {
    super(cause);
  }

  public LabzenRuntimeException(Throwable cause, String message) {
    super(message, cause);
  }

  public LabzenRuntimeException(Throwable cause, String message, Object... args) {
    super(MessageFormatter.basicArrayFormat(message, args), cause);
  }
}

File: src/main/kotlin/cn/labzen/meta/exception/LabzenRuntimeException.kt
package cn.labzen.meta.exception

import org.slf4j.helpers.MessageFormatter

abstract class LabzenRuntimeException : RuntimeException {
  constructor(message: String) : super(message)
  constructor(message: String, vararg arguments: Any?) : super(MessageFormatter.basicArrayFormat(message, arguments))

  constructor(cause: Throwable) : super(cause)
  constructor(cause: Throwable, message: String) : super(message, cause)
  constructor(cause: Throwable, message: String, vararg arguments: Any?) :
      super(MessageFormatter.basicArrayFormat(message, arguments), cause)
}
